src/components/data/DataNavigation.js

```diff
@@ -397,10 +401,12 @@ function DataNavigation(props) {
 
     const handleMenuItemClick = (event, index) => {
         setAnchorEl(null);
+        const storageId = router.pathname.split(constants.PATH_SEPARATOR)[2];
         router.push(
-            PATH_SEPARATOR +
+            constants.PATH_SEPARATOR +
                 NavigationConstants.DATA +
-                `?path=${dataRoots[index].path}`
+                constants.PATH_SEPARATOR +
+                `${storageId}${dataRoots[index].path}`
```

Comment:
I think it's preferable to construct this entirely with string interpolation in 1 template literal, rather than mixing template literals with `+` concatenation (I think there's an `eslint` rule for this as well):
```javascript
`${constants.PATH_SEPARATOR}${NavigationConstants.DATA}${constants.PATH_SEPARATOR}${storageId}${dataRoots[index].path}`
```